### DISCUSSION TITLE
Git sha for raw content

### DISCUSSION THREAD

**Discussion by namse** [community/community#24589] [Apps, API and Webhooks]
<p><code>raw.githubusercontent.com</code> doesn’t seem to give sha in response header.</p>
<p>Let’s assume user A try to get sha and download url(<a href="http://raw.githubusercontent.com" rel="noopener nofollow ugc">raw.githubusercontent.com</a>) from <a href="https://docs.github.com/en/rest/repos/contents#get-repository-content" rel="noopener nofollow ugc">get-repository-content</a> api. User A got sha and started downloading to get binary raw data.</p>
<p>If other user B push the new commit, which edit the resource which one user A try to get, just before user A started downloading, then user A may get different data against sha.</p>
<p>Is there any chance to make sure that user A downloaded right data for specific sha using API?</p>

**Comment by namse** [comment]
<p>I found that <a href="https://docs.github.com/en/rest/git/blobs#get-a-blob" rel="noopener nofollow ugc">get-a-blob</a> api support about this.</p>

**Comment by namse** [comment]
<p>What if for the file which &gt; 100 MB? get-a-blob only support &lt;= 100MB files to get content.</p>

**Comment by airtower-luna** [comment, marked as answer]
<p>GitHub blocks pushes larger than 100 MB anyway, see <a href="https://docs.github.com/en/repositories/working-with-files/managing-large-files/about-large-files-on-github" class="inline-onebox">About large files on GitHub - GitHub Docs</a>. If you want to handle larger files you’ll probably have to support Git LFS.</p>
